lib/Jsrt/Jsrt.cpp

```diff
@@ -1415,7 +1415,8 @@ CHAKRA_API JsInstanceOf(_In_ JsValueRef object, _In_ JsValueRef constructor, _Ou
         VALIDATE_INCOMING_REFERENCE(constructor, scriptContext);
         PARAM_NOT_NULL(result);
 
-        *result = Js::RecyclableObject::FromVar(constructor)->HasInstance(object, scriptContext) ? true : false;
+        Js::Var booleanValue = Js::JavascriptOperators::OP_IsInst(object, constructor, scriptContext, nullptr);
```

Comment:
I think this might not be guaranteed to return a `Js::JavascriptBoolean`? It looks like in the case of a callable `Symbol.hasInstance` that it will return the result of the function, and it is possible to define a function that doesn't return a Boolean. If I do that from script, then `foo instanceof bar` appears to return the truthiness of whatever I return, but I don't see that in the `OP_IsInst` codepath

Reply:
I think you might need to use `JavascriptConversion::ToBoolean` here, and then the next line should be safe.

Reply:
Ah; I misread the code in `IsInst` and it does in fact do the conversion there. Disregard!